schema: Add dependency additions

nilmerg opened the pull request:
As discussed in #347.

Todo:

- [ ] Remove constraints
- [ ] Rename keys
- [ ] Add missing columns (environment_id etc)
- [ ] Integrate into full schema file
- [ ] Add PostgreSQL version

raviks789 commented on schema/mysql/upgrades/dependencies.sql:
```diff
@@ -0,0 +1,57 @@
+CREATE TABLE `redundancy_group_state` (
+  `id` binary(20) NOT NULL,
+  `redundancy_group_id` binary(20) NOT NULL,
+  `failed` enum('n', 'y') NOT NULL,
```
The `redundancy_group_state` table is missing `last_state_change` column, to track how long since the redundancy group has failed.

nilmerg replied:
done

oxzi replied:
It seems that now it's there twice. Please refer to my other comment.

oxzi commented on schema/mysql/upgrades/dependencies.sql:
For my testing, I have added the environment ID as `environment_id BINARY(20) NOT NULL` to most tables.

oxzi commented on schema/mysql/upgrades/dependencies2.sql:
```diff
@@ -0,0 +1,2 @@
+alter table redundancy_group_state
+    add last_state_change bigint unsigned not null;
```
https://github.com/Icinga/icingadb/blob/dbd05f20420ed986f8d36942ca6b3582cbc8d472/schema/mysql/upgrades/dependencies.sql#L28

Unless I am missing something, it is already there.

nilmerg commented:
I've cleaned it up. `dependencies_full.sql` now contains all additions. The numbered files are partial upgrades.

Also, I've added `redundancy_group_state.is_reachable`.

julianbrost commented:
This is now part of #889 which supersedes this PR.
